src/Network/Algorand/Node/Util.hs

```diff
@@ -29,3 +40,22 @@ transactionStatus TransactionInfo{tiConfirmedRound, tiPoolError} =
     Nothing -> case T.null tiPoolError of
       False -> KickedOut tiPoolError
       True -> Waiting
+
+-- | Catches error that has block absence for the round as the cause and
+-- return Nothing in this case. Other errors would be rethrown.
+getBlock
+  :: MonadCatch m
+  => Api.ApiV2 (AsClientT m) -> B.Round -> m (Maybe B.Block)
+getBlock api rnd = handle handler $ do
+  B.BlockWrapped block <- Api._block api rnd Api.msgPackFormat
+  pure (Just block)
+  where
+    noBlockMsg = "ledger does not have entry"
+    handler (FailureResponse _req
+              Response { responseStatusCode = s, responseBody = b })
+      | statusCode s == 500
```

Comment:
OMG, they really return 500 (aka internal server error) for the case when block with given round doesn't exist. It's really weird to me because it's not an internal error, I would expect 404 or 400 or maybe another 4**. But I've even checked myself and can confirm this behavior. So that's ok on our side.